cmd/booster-http/server.go

```diff
@@ -150,32 +148,55 @@ func (s *HttpServer) handleIndex(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(idxPage)) //nolint:errcheck
 }
 
-func (s *HttpServer) handleByPayloadCid(w http.ResponseWriter, r *http.Request) {
-	startTime := time.Now()
-	ctx, span := tracing.Tracer.Start(r.Context(), "http.payload_cid")
-	defer span.End()
+func (s *HttpServer) handlePieceRequest(w http.ResponseWriter, r *http.Request) {
+	q, err := url.ParseQuery(r.URL.RawQuery)
+	if err != nil {
+		msg := fmt.Sprintf("parsing query: %s", err.Error())
+		writeError(w, r, http.StatusBadRequest, msg)
+	}
 
-	stats.Record(ctx, metrics.HttpPayloadByCidRequestCount.M(1))
+	isCar := false
 
-	// Remove the path up to the payload cid
-	prefixLen := len(s.payloadBasePath())
-	if len(r.URL.Path) <= prefixLen {
-		msg := fmt.Sprintf("path '%s' is missing payload CID", r.URL.Path)
-		writeError(w, r, http.StatusBadRequest, msg)
-		stats.Record(ctx, metrics.HttpPayloadByCid400ResponseCount.M(1))
-		return
+	if len(q["format"]) == 1 {
+		if q["format"][0] == "car" { // Check if format value is car
+			isCar = true
+		} else if q["format"][0] != "piece" { // Check if format value is not piece
+			writeError(w, r, http.StatusBadRequest, "unsupported format")
+		}
+	} else { // Error if more than 1 format value
+		writeError(w, r, http.StatusBadRequest, "unsupported query")
```

Comment:
Let's try to give the user as much information as possible if they get the parameters wrong:
```suggestion
		writeError(w, r, http.StatusBadRequest, "missing 'format' query parameter")
```

Reply:
We support no format query and default to a piece in that case. I have changed the error msg to be more detailed though.

Reply:
I'm not sure if I'm reading the code correctly, but it looks like it will return an error if the user doesn't specify the `format` query parameter.

If the user doesn't specify the `format` query parameter then `len(q["format])` will be zero here, right?
```
if len(q["format"]) == 1
```

Reply:
My bad. Fixing it.